src/masternodes/mn_checks.cpp

```diff
@@ -4019,31 +4035,35 @@ bool IsDisabledTx(uint32_t height, CustomTxType type, const Consensus::Params &c
         }
     }
 
-    // ICXCreateOrder      = '1',
-    // ICXMakeOffer        = '2',
-    // ICXSubmitDFCHTLC    = '3',
-    // ICXSubmitEXTHTLC    = '4',
-    // ICXClaimDFCHTLC     = '5',
-    // ICXCloseOrder       = '6',
-    // ICXCloseOffer       = '7',
-
-    // disable ICX orders for all networks other than testnet
-    if (Params().NetworkIDString() == CBaseChainParams::REGTEST ||
-        (IsTestNetwork() && static_cast<int>(height) >= 1250000)) {
-        return false;
-    }
-
-    // Leaving close orders, as withdrawal of existing should be ok
-    switch (type) {
-        case CustomTxType::ICXCreateOrder:
-        case CustomTxType::ICXMakeOffer:
-        case CustomTxType::ICXSubmitDFCHTLC:
-        case CustomTxType::ICXSubmitEXTHTLC:
-        case CustomTxType::ICXClaimDFCHTLC:
-            return true;
-        default:
+    if (height < Params().GetConsensus().NextNetworkUpgradeHeight) {
+        // ICXCreateOrder      = '1',
+        // ICXMakeOffer        = '2',
+        // ICXSubmitDFCHTLC    = '3',
+        // ICXSubmitEXTHTLC    = '4',
+        // ICXClaimDFCHTLC     = '5',
+        // ICXCloseOrder       = '6',
+        // ICXCloseOffer       = '7',
+
+        // disable ICX orders for all networks other than testnet
+        if (Params().NetworkIDString() == CBaseChainParams::REGTEST ||
```

Comment:
Let's turn it such that this checks the `IsICXBuggedRegion()` region. And it's a single source of check.